posting.go

```diff
@@ -355,6 +355,12 @@ func (i *PostingsIterator) readFreqNormHasLocs() (uint64, uint64, bool, error) {
 		return 1, i.normBits1Hit, false, nil
 	}
 
+	// check if freq/norm is disabled.
+	if i.freqNormReader.isNil() {
+		// locations could be there even without freq/norm.
```

Comment:
it's not clear to me from this comment, that this case is handled, or if this comment is worrying about that possibility

Reply:
also, it seems I'm late to the game, but the `isNil()` method doens't feel very idiomatic

Reply:
When there is no freq/norm info available, then the readFreqNormHasLocs () call just does a NOOP and let the callers handle any further processing of location details is the intent here.

isNil()  was just using an already existing method on the intDecoder.

Reply:
> Hmmn, still need to think about part of this.
> 
> Also, what is the effective space savings? It seems like we still need to record frequency of 0. Is that a varint? So, do we only save not writing out the norm?

When disabled, we don't write any freq/norm info to disk except the `termNotEncoded` offset value here- https://github.com/blevesearch/zapx/blob/master/intcoder.go#L160

Yes, those are varints - https://github.com/blevesearch/zapx/blob/master/merge.go#L618

Reply:
Savings were roughly like 10% or so in a default index of travel-sample with hardcoded freq/norm disabled with all other default options kept as it is.